src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -762,7 +762,7 @@ where
         while !cap_set.is_empty() {
             // AsyncInputHandle::next is cancel safe
             tokio::select! {
-                _ = ticker.tick() => {
+                _ = ticker.tick(), if *source_upper.frontier() != [FromTime::minimum()] => {
```

Comment:
what this `if` guard guarantees and its importance should be documented (maybe on the `mint` fn itself?)